build/relsignDSCModules.sh

```diff
@@ -0,0 +1,309 @@
+# It generates a checksum file for each DSC module, uploads to azure storage account using SAS_SOURCE_URI,
+# then downloads the signed *.asc file from SAS_DEST_URI and 
+# serializes it into the zip package for given DSC module.
+
+StorageVersion="2014-02-14"
+# Max number of attempts to wait for signing worklow to finish 
+MaxSignWaitAttempts=30
+
+# read_blob <URI> <REMOTE_FILE> <DEST_FILE>
+read_blob() {
+
+    [ ! -z "$1" ] || error "read_blob(): expected remote URI" 1
+    [ ! -z "$2" ] || error "read_blob(): expected remote filename" 1
+    [ ! -z "$3" ] || error "read_blob(): expected destination file" 1
+
+    url="$1"
+    remotefile="$2"
+    localfile="$3"
+
+    container=$( echo "$1" | awk -F \?  '{container=index($0,"?"); print $1}' )
+	echo "container is $container"
+    sas=$( echo "$url" | awk -F \? '{container=index($0,"?"); print $2}' )
+    url="${container}/${remotefile}?${sas}"
+
+    timestamp=$(date +%Y-%m-%dT%H:%M:%SZ)
+    output=$(curl --silent --write-out %{http_code} --request "GET" \
+                  --header "x-ms-date    : $timestamp" \
+                  --header "x-ms-version : $StorageVersion" \
+                  "$url" )
+    
+    code=${output: (-3)}	
+	if [ "$code" != 200 ]; then
+	    echo "read_blob(): error retrieving $remotefile, server returned an http error code $code"
+        return 1
+    fi
+
+    output=${output: :(-3)}
+    echo -n "$output" > "$localfile"
+   
+    return 0
+}
+
+# write_blob <SASURL> <FILE>
+write_blob() {
+
+    [ ! -z "$1" ] || error "write_blob(): expected remote URI" 1
+    [ ! -z "$2" ] || error "write_blob(): expected filename to upload" 1
+    [ -f "$2" ] || error "write_blob(): file $2 does not exist" 1
+
+    url="$1"
+    localfile="$2"
+    remotefile=$(basename "$localfile")
+
+    container=$( echo "$1" | awk -F \?  '{container=index($0,"?"); print $1}' )
+	echo "container is $container"
+    sas=$( echo "$url" | awk -F \? '{container=index($0,"?"); print $2}' )
+    url="${container}/${remotefile}?${sas}"
+
+    timestamp=$(date +%Y-%m-%dT%H:%M:%SZ)
+    output=$(curl --silent --write-out %{http_code} --request "PUT" \
+                  --header "x-ms-date      : $timestamp" \
+                  --header "x-ms-blob-type : BlockBlob" \
+                  --header "x-ms-version   : $StorageVersion" \
+                  --data-binary "@${localfile}" "$url" )
+    
+    code=${output: (-3)}
+	if [ "$code" != 201 ]; then
+        echo "write_blob(): error writing $remotefile, server returned an http error code $code"
+		return 1
+    fi
+
+    return 0
+}
+
+# delete <URI> <REMOTE_FILE>
+delete_blob() {
+
+    [ ! -z "$1" ] || error "delete_blob(): expected remote URI" 1
+    [ ! -z "$2" ] || error "delete_blob(): expected remote filename" 1
+   
+    url="$1"
+    remotefile="$2"
+    
+    container=$( echo "$1" | awk -F \?  '{container=index($0,"?"); print $1}' )
+	echo "container is $container"
```

Comment:
Inconsistent indentation due to tabs. I'm okay with this (it shows up properly in Emacs or some reasonable editor). But just FYI, we often tell our editors to space indent to avoid indentation issues across a variety of editors (particularly this on Windows).